azurelinuxagent/ga/monitor.py

```diff
@@ -142,24 +142,23 @@ def __init__(self):
         self.imds_errorstate = ErrorState(min_timedelta=MonitorHandler.IMDS_HEALTH_PERIOD)
 
     def run(self):
-        self.init_protocols()
-        self.init_sysinfo()
-        self.start()
+        self.start(init_data=True)
 
     def stop(self):
         self.should_run = False
         if self.is_alive():
             self.event_thread.join()
 
     def init_protocols(self):
+        self.protocol_util = get_protocol_util()
```

Comment:
I wanted to remove this property altogether but didnt mainly because a lot of test cases are dependent on this. I can take this up as a separate PR once I get back from vacation